test/smoke_test/bcda_client.go

```diff
@@ -203,22 +225,96 @@ func main() {
 
 					if encrypt {
 						fmt.Println("decrypting the file...")
-						encryptedKey, err := hex.DecodeString(encryptData[path.Base(data[0].Url)])
-						if err != nil {
-							panic(err)
-						}
+						encryptedKey := string(encryptData[path.Base(data[0].Url)])
+
 						privateKeyFile := os.Getenv("ATO_PRIVATE_KEY_FILE")
-						privateKey := getPrivateKey(privateKeyFile)
-						filename = decryptFile(privateKey, encryptedKey, filename)
-						fmt.Printf("writing decrypted file to disk: %s\n", filename)
-					}
 
-					fmt.Println("validating file content...")
-					if !isValidNDJSONFile(filename) {
-						fmt.Println("Error: file is not in valid NDJSON format!")
-						os.Exit(1)
-					}
+						// execute the golang decryptor
+						var cmdOut []byte
+						cmdName := "go"
+						cmdArgs := []string{
+							"run", "../../decryption_utils/Go/decrypt.go",
+							"--file", filename,
+							"--pk", privateKeyFile,
+							"--key", encryptedKey,
+						}
+						fmt.Println("Running the go decryptor externally...")
+						// #nosec
+						if cmdOut, err = exec.Command(cmdName, cmdArgs...).Output(); err != nil {
+							fmt.Fprintln(os.Stderr, "There was an error running the go decryption util: ", err)
+							os.Exit(1)
+						}
+						output := string(cmdOut)
+
+						fmt.Println("validating Go decryptor content...")
+						if !isValidNDJSONText(output) {
+							fmt.Println("Error: file is not in valid NDJSON format!")
+							os.Exit(1)
+						}
+
+						// execute the Python decryptor
+						cmdName = "python"
+						cmdArgs = []string{
+							"../../decryption_utils/Python/decrypt.py",
+							"--file", filename,
+							"--pk", privateKeyFile,
+							"--key", string(encryptedKey),
+						}
+						fmt.Println("Running the Python decryptor externally...")
+						fmt.Println(cmdArgs)
+						// #nosec
+						if cmdOut, err = exec.Command(cmdName, cmdArgs...).Output(); err != nil {
+							fmt.Fprintln(os.Stderr, "There was an error running the Python decryption util: ", err)
+							os.Exit(1)
+						}
+
+						output = string(cmdOut)
+
+						fmt.Println("validating Python decryptor content...")
+						if !isValidNDJSONText(output) {
+							fmt.Println("Error: file is not in valid NDJSON format!")
+							os.Exit(1)
+						}
+
+						// execute the C# decryptor
+						cmdName = "dotnet"
+						cmdArgs = []string{"run",
+							"--project", "../../decryption_utils/C#",
+							"decrypt.cs",
+							"--file", filename,
+							"--pk", privateKeyFile,
+							"--key", encryptedKey,
+						}
+
+						fmt.Println("Running the C# decryptor externally...")
+
+						// C# puts a bunch of nonsense in the file the first time it runs so this run is just ignored.
```

Comment:
Does the extra content come 1) the first time it's run after installing, 2) the first time in a smoke test, or 3) the first of every pair of times it's running now?  I'd guess it's 1), and that running it once with an empty argument list after invoking the installation script would be sufficient.

Reply:
It is 1.  I was trying to run it earlier in the process, but I got some non zero return codes and it aborted.  It def adds a few seconds to the run, but it shouldn't really be a problem to do this twice.